v2/tools/generator/internal/astmodel/package_definition.go

```diff
@@ -58,12 +59,21 @@ func (p *PackageDefinition) EmitDefinitions(outputDir string, generatedPackages
 		return 0, err
 	}
 
-	// Only generate GroupVersion file if this file includes resources
+	// Check if current package contains resources
 	if resources := FindResourceDefinitions(p.definitions); len(resources) > 0 {
+
+		// If package contains resources, then we generate GroupVersion file for the package
 		err = emitGroupVersionFile(p, outputDir)
 		if err != nil {
 			return 0, err
 		}
+
+		// If package path does not include crossplane api dir and is not Storage package, then we generate doc file for the package
+		if !strings.HasSuffix(p.PackageName, StoragePackageSuffix) && !strings.Contains(outputDir, "apis") {
```

Comment:
Maybe rather than specifically checking for crossplane here, you can make a config option in `azure-arm.yaml` that sets `emitdocs` to true, and then check that here? That way crossplane's YAML can set it to false (or not set it at all and false is the default) and `azure-arm.yaml` can set it to true, and we don't need to hardcode understanding that there are two into the actual tool.